docs/reference/migration/migrate_7_0/migrate_to_java_time.asciidoc

```diff
@@ -0,0 +1,309 @@
+[[migrate_to_java_time]]
+=== Java Time Migration Guide
+
+Since version 7.0 Elasticsearch is using java.time as an underlying implementation of dates parsing, formatting and calculations.
+java.time is in many aspects very similar to joda-time which we previously used. However there are some differences and this guide is meant to
+help you to become compatible.
+
+==== Identifying if you are affected
+You are only affected if you have used any custom date formats in your mappings, pipelines or templates.
```

Comment:
The switch to java time only impacts custom date formats in the following features:

* <<mapping,Index mappings>>
* <<indices-templates,Index templates>>
* <<pipeline,Ingest pipelines>>

If you don't use custom date formats, you can skip the rest of this guide.